### PR TITLE
ticket services hierarchy

### PR DESCRIPTION
**Pull request by ayazbayainur**

### CODE REVIEW

File: AndersenHWs/src/main/java/org/example/Admin.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.example;
+
+public class Admin extends User{
+    @Override
+    public Role printRole(){
+        return Role.ADMIN;
+    }
+
+    private boolean checkTicket(Ticket ticket){
```
Comment (Alexmansar) [inline AndersenHWs/src/main/java/org/example/Admin.java]: should be public)

File: AndersenHWs/src/main/java/org/example/Client.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.example;
+
+public class Client extends User{
+    @Override
+    public Role printRole(){
+        return Role.CLIENT;
+    }
+
+    private Ticket getTicket(long id){
```
Comment (Alexmansar) [inline AndersenHWs/src/main/java/org/example/Client.java]: same

File: AndersenHWs/src/main/java/org/example/ConcertTicket.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
+package org.example;
+
+import java.util.Objects;
+
+public class ConcertTicket extends Ticket{
+    public static Double price;
```
Comment (Alexmansar) [inline AndersenHWs/src/main/java/org/example/ConcertTicket.java]: hy its pubic static, not just private as other ones?

File: AndersenHWs/src/main/java/org/example/ConcertTicket.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    public ConcertTicket shared(String phone, String email){
+        if(this.phone == phone && this.email == email){
+            return this;
+        }
+        return null;
+    }
+
+    public String toString(){
```
Comment (Alexmansar) [inline AndersenHWs/src/main/java/org/example/ConcertTicket.java:108]: not critical, but you can add Override to show that we update Object method))

File: AndersenHWs/src/main/java/org/example/Ticket.java [hunk 5]
```diff
@@ -1,100 +1,16 @@
...
+    public String print(){
+        return "print content in console";
     }
 
-    enum StadiumSector{
-        A, B, C;
+    public Ticket shared(String phone){
+        return null;
     }
+    public abstract Ticket shared(String phone, String email);
```
Comment (Alexmansar) [inline AndersenHWs/src/main/java/org/example/Ticket.java]: could be void and just print some message)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@ayazbayainur Hi)) fix small comments))

File: AndersenHWs/src/main/java/org/example/ConcertTicket.java (hunk 3 above)
Comment (ayazbayainur) [reply AndersenHWs/src/main/java/org/example/ConcertTicket.java]: *** Add the ability to save ticket price. This should be a new variable, think carefully about the type.

I thought that public static would do the job. In the Ticket Service class, I wrote "ConcertTicket.price = 3000.0;", and thought that making price static -  the ability to save ticket price.

**Review by Alexmansar — APPROVED** [review]
@ayazbayainur Hi, could me merged)

**Pull request merged**